src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NonConstantFieldsShouldNotBeVisible.cs

```diff
@@ -32,8 +33,23 @@ public abstract class NonConstantFieldsShouldNotBeVisibleAnalyzer : DiagnosticAn
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
-        { 
-            
+        {
+            analysisContext.RegisterSymbolAction(obj =>
+            {
+                var field = (IFieldSymbol)obj.Symbol;
+
+                if (!field.IsPublic())
+                {
+                    return;
+                }
+
+                if (field.IsConst || (field.IsStatic && field.IsReadOnly))
+                {
+                    return;
+                }
+
+                obj.ReportDiagnostic(field.CreateDiagnostic(Rule));
```

Comment:
I don't think a field shows up as const in IL. Does the value gets inlined by the compiler? If so, FXCop wouldn't have needed to look for that.